### PR TITLE
Refactor modularization code to avoid nested functions. NFC

### PR DESCRIPTION
**Pull request by sbc100**
This also avoids the needs for an IIFE here, which makes the output code much readable.

This also saves on code size.

### CODE REVIEW

File: tools/link.py [hunk 1]
```diff
@@ -2325,23 +2325,15 @@ def modularize():
...
-    'maybe_async': async_emit,
-    'src': src,
-    'return_value': return_value,
-  }
-
   if settings.MINIMAL_RUNTIME and not settings.PTHREADS:
     # Single threaded MINIMAL_RUNTIME programs do not need access to
     # document.currentScript, so a simple export declaration is enough.
-    src = '/** @nocollapse */ var %s = %s' % (settings.EXPORT_NAME, src)
+    src = f'''\
```
Comment (sbc100) [inline tools/link.py:2331]: I'm not sure why @nocollapse was added here.  That seems to only apply to object or to object properties, not to function.

Comment (sbc100) [reply tools/link.py:2331]: https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#nocollapse

File: tools/link.py [hunk 2]
```diff
@@ -2325,23 +2325,15 @@ def modularize():
...
+var {settings.EXPORT_NAME} = {async_emit}(moduleArg = {{}}) => {{
```
Comment (kripken) [inline tools/link.py:2332]: What is `{{}}` here?

Comment (sbc100) [reply tools/link.py:2332]: `{{` and `}}` are used to escape `{` and `}` in python f-strings

**Review by kripken — APPROVED** [review]

**Pull request merged**